### PR TITLE
ReportCsvWriter

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1026

### Change description ###
ReportCsvWriter which will be called while sending reports to email.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -224,6 +224,8 @@ dependencies {
   implementation group: 'uk.gov.hmcts.reform', name: 'logging-appinsights', version: versions.reformLogging
 
   implementation group: 'uk.gov.hmcts.reform', name: 'properties-volume-spring-boot-starter', version:'0.0.4'
+  implementation group: 'org.apache.commons', name: 'commons-csv', version: '1.8'
+  implementation group: 'org.apache.commons', name: 'commons-collections4', version: '4.1'
```
Comment (doncem) [inline build.gradle]: is extra dependency really necessary?

Comment (apereverzin) [reply build.gradle]: it is. in the processor it is transitive via   compile group: 'com.microsoft.azure', name: 'azure-servicebus', version: '1.2.18' but in this case I have decided it should be an overkill

Comment (doncem) [reply build.gradle]: yea, but is it really necessary? Only used for `CollectionUtils.emptyIfNull`. Can make sure any other way like `Optoinal.ofNullable` for starters. But in general, the incoming `List` should not be empty aka null in first place. Losing the point of being a list

Comment (apereverzin) [reply build.gradle]: have followed the advice

**Pull request merged**